pivot.py

```diff
@@ -0,0 +1,1723 @@
+import re
+from werkzeug.routing import Rule
+from werkzeug.utils import redirect, send_from_directory
+from dominate.tags import (div, h1, h2, p, a, form, button, span, table, thead,
+    tbody, tr, td, input_, br, head, html, body, meta, link, title, script, h3,
+    comment, section, nav, ul, li, img, footer, label, ol, dl, dt, dd,
+    select, option, main, th, fieldset, legend, h4, time_, h5, aside)
+from dominate.util import raw
+from trytond.model import fields
+from trytond.pool import Pool, PoolMeta
+from trytond.transaction import Transaction
+from trytond.modules.voyager.voyager import Component, Trigger
+from trytond.modules.voyager.i18n import _
+from werkzeug.wrappers import Response
+
+from collections import deque
+
+COLLAPSE = '➖'
+EXPAND = '➕'
+
+class Site(metaclass=PoolMeta):
+    __name__ = 'www.site'
+
+    @classmethod
+    def __setup__(cls):
+        super().__setup__()
+        cls.type.selection += [('babi_pivot', 'Pivot')]
+
+###############################################################################
+################################ PIVOT CLASSES ################################
+###############################################################################
+class Operation:
```

Comment:
I suspect we should try to better separate HTML generation from table generation